packages/platform-android/src/commands/runAndroid/index.ts

```diff
@@ -135,34 +135,74 @@ function buildAndRun(args: Flags) {
     packageName,
   );
   const adbPath = getAdbPath();
-  if (args.deviceId) {
+  let devices;
+
+  if (!args.deviceId) {
+    devices = adb.getDevices(adbPath);
+  }
+
+  if (
+    devices &&
+    devices.length > 1 &&
+    devices.filter(Boolean).length > 1 &&
+    args.interactive
+  ) {
+    devices = await chooseDevice(devices);
+  }
+
+  if (args.deviceId || (devices && devices.length === 1)) {
     return runOnSpecificDevice(
       args,
       cmd,
       packageNameWithSuffix,
       packageName,
       adbPath,
+      devices,
```

Comment:
If the function is `runOnSpecificDevice`, wouldn't it make more sense to pass just `device`? (not the entire array)? We could reduce the logic inside the function.